### PR TITLE
#47-Login-Page

### PR DESCRIPTION
**Pull request by AsmaMohame**

### CODE REVIEW

File: retail-managment-system/rms-ui/src/app/core/layout/header/header.component.html [hunk 1]
```diff
@@ -9,9 +9,12 @@
     >
 
     <div class="collapse navbar-collapse">
-      <button class="btn btn-outline-light ms-auto" type="submit">
+      <button class="btn btn-outline-light ms-auto" type="submit" (click)="login()">
         {{ "login" | translate }}
       </button>
+      <button class="btn btn-outline-light ms-3" type="submit" (click)="logout()">
+        Log out
+      </button>
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/core/layout/header/header.component.html:17]: login or logout appear depend on user logged on or not

**Review by ZinabHashem — CHANGES_REQUESTED** [review]

File: retail-managment-system/rms-ui/src/app/pages/login/login.component.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      if(this.loginUsers.find(u=>u.email==user.email))
+         {
+           if(this.loginUsers.find(u=>u.password==user.password))
+           {
+            localStorage.setItem("isLogged","Userlogged");
+            this.router.navigateByUrl('/home');
+           }
+           else
+           {
+             alert("Password Is Incorrect")
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/login/login.component.ts:58]: use alert difference of this ex: angular material alert

**Review by ZinabHashem — CHANGES_REQUESTED** [review]

File: retail-managment-system/rms-ui/src/app/pages/login/login.component.scss [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  .button-row a {
+    margin-right: 8px;
+    text-align: center;
+  }
+  
+  .forget-password{
+    padding-left: 0px;
+  }
+  
+  .emailInput{
```
[the remark this answers is not in the document]
Comment (AsmaMohame) [reply retail-managment-system/rms-ui/src/app/pages/login/login.component.scss]: done

File: retail-managment-system/rms-ui/src/app/pages/login/login.component.scss [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    padding-top: 10px;
+  }
+  
+  .contentBody {
```
[the remark this answers is not in the document]
Comment (AsmaMohame) [reply retail-managment-system/rms-ui/src/app/pages/login/login.component.scss]: I remove it

File: retail-managment-system/rms-ui/src/app/pages/login/login.component.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      {email:"raghda@gmail.com", password:"666"}
+    ]
+  }
+
+  ngOnInit(): void {
+    this.logForm();
+    this.setUsers();
+  }
+
+  logForm() {
```
[the remark this answers is not in the document]
Comment (AsmaMohame) [reply retail-managment-system/rms-ui/src/app/pages/login/login.component.ts]: done

File: retail-managment-system/rms-ui/src/app/pages/pages.module.ts [hunk 6]
```diff
@@ -34,6 +36,7 @@ const routes: Routes = [
     ProfitComponent,
     InvestorPageComponent,
     InstallmentComponent,
+    LoginComponent
```
[the remark this answers is not in the document]
Comment (AsmaMohame) [reply retail-managment-system/rms-ui/src/app/pages/pages.module.ts]: done

File: retail-managment-system/rms-ui/src/app/core/layout/header/header.component.html (hunk 1 above)
Comment (AsmaMohame) [reply retail-managment-system/rms-ui/src/app/core/layout/header/header.component.html:17]: done

File: retail-managment-system/rms-ui/src/app/pages/pages.module.ts [hunk 7]
```diff
@@ -34,8 +37,9 @@ const routes: Routes = [
     ProfitComponent,
     InvestorPageComponent,
     InstallmentComponent,
+   
   ],
 
-  imports: [RouterModule.forChild(routes), SharedModule, TranslateModule],
+  imports: [RouterModule.forChild(routes), SharedModule, TranslateModule,AuthModule],
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/pages.module.ts:43]: why import AuthModule here?

File: retail-managment-system/rms-ui/src/app/core/layout/header/header.component.ts [hunk 8]
```diff
@@ -11,9 +12,11 @@ export class HeaderComponent {
   @Input() sideNavItem!: MatSidenav;
   lang: string = 'en';
 
-  constructor(private router: Router, private translate: TranslateService) {
+  constructor(private router: Router, private translate: TranslateService ,) {
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/core/layout/header/header.component.ts:15]: delete extra comma

File: retail-managment-system/rms-ui/src/app/core/layout/header/header.component.ts [hunk 9]
```diff
@@ -28,4 +31,15 @@ export class HeaderComponent {
...
     window.location.reload();
   }
+
+  logout()
+  {
+    localStorage.removeItem("isLogged")
+    this.router.navigateByUrl('/login')
+  }
+
+  get userLogged () :boolean
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/core/layout/header/header.component.ts:41]: make it as normal function and set true and false in property

File: retail-managment-system/rms-ui/src/app/auth/components/login/login.component.scss [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    padding-top: 5px;
+  }
+  
+  .button-row a {
+    margin-right: 8px;
+    text-align: center;
+  }
+  
+  .forget-password{
+    padding-left: 0px;
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/auth/components/login/login.component.scss:28]: there is this style in bootstrap why didn't used it?
and some others in this file.

**Review by ZinabHashem — CHANGES_REQUESTED** [review]

File: retail-managment-system/rms-ui/src/app/auth/components/login/login.component.ts [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+         {
+           if(this.loginUsers.find(u=>u.password==user.password))
+           {
+            localStorage.setItem("isLogged","Userlogged");
+            this.router.navigateByUrl('/home');
+           
+           }
+           else
+           {
+            this._snackBar.open('Password is Not correct!', 'Close', {
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/auth/components/login/login.component.ts:63]: translate error messages too

File: retail-managment-system/rms-ui/src/app/auth/auth.module.ts [hunk 12]
```diff
@@ -1,8 +1,15 @@
...
+import { TranslateModule } from '@ngx-translate/core';
 
+const routes: Routes = [
+  { path: 'login', component: LoginComponent},
+]
 @NgModule({
-  declarations: [],
-  imports: [CommonModule],
+  declarations: [LoginComponent],
+  imports: [RouterModule.forChild(routes) ,CommonModule , SharedModule , RouterModule,TranslateModule],
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/auth/auth.module.ts:13]: why duplicate RouterModule?

File: retail-managment-system/rms-ui/src/app/auth/components/login/login.component.scss [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+    max-height: 600px;
+  }
+  .main {
+    height: 630px;
+  }
+  
+  .header{
+    text-align: center;
+    font-weight: bold;
+    font-size: large;
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/auth/components/login/login.component.scss:14]: text-align: center;
    font-weight: bold;
    font-size: large;
   --->  are this exist in bootstrap or not?
    and any thing like this in file css

**Review by ZinabHashem — CHANGES_REQUESTED** [review]

File: retail-managment-system/rms-ui/src/app/auth/components/login/login.component.ts [hunk 14]
```diff
@@ -69,14 +75,21 @@ export class LoginComponent implements OnInit {
...
            return;
          }
     }
   }
 
+  translateError ()
+  {
+    this.emailError= this.translate.instant('logIn.email-error');
+    this.passwordError= this.translate.instant('logIn.password-error');
+    this.close= this.translate.instant('logIn.close');
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/auth/components/login/login.component.ts:90]: you can use  this.translate.instant('logIn.close') and others in place you want without do function for that

**Pull request merged**